### PR TITLE
fix unhandled edge cases for keyboard shortcuts (fixes #927, fixes #932, fixes #1063)

### LINKED ISSUE #927: WASD controls should stop on document blur
I noticed this a while back but kept forgetting to file.

STR:

1. Load https://aframe.io/aframe/examples/showcase-spheres-and-fog/ and stay in mono view
2. Hold down the `A` key
3. Whilst holding the `A` key, move your cursor outside of your browser window
4. Notice you are still moving left

### LINKED ISSUE #932: WASD controls should not be engaged w/ shift/control/alt/meta keys
(Similar but different from #927.)

STR:

1. Load https://aframe.io/aframe/examples/showcase-spheres-and-fog/
2. Hold down the `Command` + `A` keys
3. Release the keys
4. Notice you keep moving left

For WASD controls, we should abort if `shiftKey`, `ctrlKey`, `altKey`, or `metaKey` are also being held.

### LINKED ISSUE #1063: Keypresses in form fields should not be considered keypresses for keyboard shortcuts + WASD controls
1. Place a form field somewhere on the page (outside of the `<a-scene>` of course): `<input style="position:absolute;z-index:1">`
2. Start typing these keys: `w`, `a`, `s`, `d`, `f`, `z`

### PR DESCRIPTION
**Pull request by cvan**

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -129,14 +129,25 @@ var isIOS = module.exports.isIOS = function () {
...
  */
 module.exports.isLandscape = function () {
   return window.orientation === 90 || window.orientation === -90;
 };
 
 /**
+ * Returns whether we can safely listen to this keyboard event.
+ * @returns {Boolean} True if we can.
+ */
+module.exports.isSafeKeyEvent = function (event) {
```
Comment (dmarcos) [inline src/utils/index.js]: safe is not really the word I would use for this function. What about ```isModifierKeyEvent```. Should this be a private function of wasd component?

File: src/components/wasd-controls.js [hunk 2]
```diff
@@ -114,21 +116,29 @@ module.exports.Component = registerComponent('wasd-controls', {
...
 
   removeEventListeners: function () {
     // Keyboard events
+    window.removeEventListener('blur', this.onBlur);
     window.removeEventListener('keydown', this.onKeyDown);
     window.removeEventListener('keyup', this.onKeyUp);
   },
 
+  onBlur: function (event) {
+    this.keys = {};
```
Comment (dmarcos) [inline src/components/wasd-controls.js]: Should we just call ```this.pause()``` on blur?

File: src/utils/index.js (hunk 1 above)
Comment (dmarcos) [inline src/utils/index.js]: Oh I see you use this in multiple components.

**Comment by dmarcos** [comment]
Let's finish this.

**Comment by cvan** [comment]
yep, going to after I finish the WebVR 1.0 API work. I'm quite close, just one last Firefox for Android viewport issue.

File: src/components/wasd-controls.js (hunk 2 above)
Comment (cvan) [inline src/components/wasd-controls.js]: that doesn't fix it

**Comment by cvan** [comment]
closing in favour of a new PR against `master` instead of against `dev`

**Pull request closed without merging**